File: src/my_cpp_pkg/src/add_two_ints_client_no_oop.cpp
#include <chrono>

#include "example_interfaces/srv/add_two_ints.hpp"
#include "rclcpp/rclcpp.hpp"

int main(int argc, char** argv) {
    rclcpp::init(argc, argv);
    auto node = std::make_shared<rclcpp::Node>("add_two_ints_client_no_opp");

    auto client = node->create_client<example_interfaces::srv::AddTwoInts>(
        "add_two_ints");

    while (!client->wait_for_service(std::chrono::seconds(1))) {
        RCLCPP_WARN(node->get_logger(), "Waiting for server...");
    }

    auto request =
        std::make_shared<example_interfaces::srv::AddTwoInts::Request>();
    request->a = 9;
    request->b = 2;

    auto future = client->async_send_request(request);
    if (rclcpp::spin_until_future_complete(node, future) ==
        rclcpp::FutureReturnCode::SUCCESS) {
        RCLCPP_INFO(node->get_logger(), " %ld + %ld = %ld", request->a,
                    request->b, future.get()->sum);
    } else {
        RCLCPP_ERROR(node->get_logger(), "Error.");
    }

    rclcpp::shutdown();
    return 0;
}

File: src/my_py_pkg/my_py_pkg/add_two_ints_client_no_oop.py
#! /usr/bin/python3

import rclpy
from rclpy.node import Node
from example_interfaces.srv import AddTwoInts


def main(args=None):
    rclpy.init(args=args)
    node = Node("add_two_ints_no_oop")

    client = node.create_client(AddTwoInts, "add_two_ints")
    while not client.wait_for_service(1.0):
        node.get_logger().warn("Waiting for server...")

    request = AddTwoInts.Request()
    request.a = 2
    request.b = 5
    future = client.call_async(request=request)
    rclpy.spin_until_future_complete(node=node, future=future)

    try:
        response = future.result()
        node.get_logger().info(f"{request.a}+{request.b}={response.sum}")
    except Exception as e:
        node.get_logger().error(f"Error occured {e}")

    rclpy.shutdown()


if __name__ == "__main__":
    main()
